### PR TITLE
feat(appmesh): eagerly validate `healthCheck` settings

### PR DESCRIPTION
**Pull request by nmussy**
* Throw if healthCheck numeric thresholds aren't met
* Throw if healthCheck path is set with Protocol.TCP (previously silently caught)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts [hunk 1]
```diff
@@ -148,9 +148,27 @@ abstract class VirtualNodeBase extends cdk.Resource implements IVirtualNode {
   }
 }
 
+/**
+ * Minimum and maximum thresholds for HeathCheck numeric properties
+ *
+ * @see https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_HealthCheckPolicy.html
+ */
+const HEALTH_CHECK_PROPERTY_THRESHOLDS: {[key in (keyof CfnVirtualNode.HealthCheckProperty)]?: [number, number]} = {
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts]: Love the use of TS typing features here.

File: packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts [hunk 2]
```diff
@@ -159,6 +177,22 @@ function renderHealthCheck(hc: HealthCheck | undefined, pm: PortMapping): CfnVir
     timeoutMillis: (hc.timeout || cdk.Duration.seconds(2)).toMilliseconds(),
     unhealthyThreshold: hc.unhealthyThreshold || 2,
   };
+
+  (Object.keys(healthCheck) as Array<keyof CfnVirtualNode.HealthCheckProperty>)
+    .filter((key) => HEALTH_CHECK_PROPERTY_THRESHOLDS[key] && typeof healthCheck[key] === 'number')
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts]: You probably also want to filter for `!Token.isUnresolved`.

File: packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts (hunk 1 above)
Comment (nmussy) [reply packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts]: Thanks ❤️

File: packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts (hunk 2 above)
Comment (nmussy) [reply packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts]: Wouldn't `typeof healthCheck[key] === 'number'` cover it?

Comment (nmussy) [reply packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts]: I guess `healthCheck` could be unresolved, but I'd have to check it earlier

Comment (rix0rrr) [reply packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts]: No, Tokens can also be encoded into numbers. It will look like `-1.358238248E-18`.

Comment (nmussy) [reply packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts]: Got it, thanks. Done

**Pull request merged**